src/registrar/utility/csv_export.py

```diff
@@ -782,30 +708,325 @@ def export_data_unmanaged_domains_to_csv(csv_file, start_date, end_date):
+
+        # create a dictionary of fields which can be included in output.
+        # "extra_fields" are precomputed fields (generated in the DB or parsed).
+        FIELDS = {
+            # Parsed fields - defined above.
+            "Domain request": requested_domain_name,
+            "Region": region,
+            "Status": status_display,
+            "Election office": human_readable_election_board,
+            "Federal type": human_readable_federal_type,
+            "Domain type": human_readable_org_type,
+            "Request additional details": additional_details,
+            # Annotated fields - passed into the request dict.
+            "Creator approved domains count": request.get("creator_approved_domains_count", 0),
+            "Creator active requests count": request.get("creator_active_requests_count", 0),
+            "Alternative domains": request.get("all_alternative_domains"),
+            "Other contacts": request.get("all_other_contacts"),
+            "Current websites": request.get("all_current_websites"),
+            # Untouched FK fields - passed into the request dict.
+            "Federal agency": request.get("federal_agency__agency"),
+            "AO first name": request.get("authorizing_official__first_name"),
+            "AO last name": request.get("authorizing_official__last_name"),
+            "AO email": request.get("authorizing_official__email"),
+            "AO title/role": request.get("authorizing_official__title"),
+            "Creator first name": request.get("creator__first_name"),
+            "Creator last name": request.get("creator__last_name"),
+            "Creator email": request.get("creator__email"),
+            "Investigator": request.get("investigator__email"),
+            # Untouched fields
+            "Organization name": request.get("organization_name"),
+            "City": request.get("city"),
+            "State/territory": request.get("state_territory"),
+            "Request purpose": request.get("purpose"),
+            "CISA regional representative": request.get("cisa_representative_email"),
+            "Submitted at": request.get("submission_date"),
+        }
+
+        row = [FIELDS.get(column, "") for column in columns]
+        return row
+
+    @classmethod
+    def annotate_and_retrieve_fields(cls, requests, annotations, additional_values=None) -> QuerySet:
+        """
+        Applies annotations to a queryset and retrieves specified fields,
+        including class-defined and annotation-defined.
+
+        Parameters:
+            requests (QuerySet): Initial queryset.
+            annotations (dict, optional): Fields to compute {field_name: expression}.
+            additional_values (list, optional): Extra fields to retrieve; defaults to annotation keys if None.
+
+        Returns:
+            QuerySet: Contains dictionaries with the specified fields for each record.
+        """
+
+        if annotations is None:
```

Comment:
The logic checking for annotations is hard to read. I can't follow the different scenarios between this and line 985.